tap_mysql/sync_strategies/binlog.py

```diff
@@ -155,8 +158,12 @@ def row_to_singer_record(catalog_entry, version, db_column_map, row, time_extrac
             row_to_persist[column_name] = val.isoformat() + 'T00:00:00+00:00'
 
         elif isinstance(val, datetime.timedelta):
-            timedelta_from_epoch = datetime.datetime.utcfromtimestamp(0) + val
-            row_to_persist[column_name] = timedelta_from_epoch.isoformat() + '+00:00'
+            if property_format == 'time':
+                # this should convert time column into 'HH:MM:SS' formatted string
+                row_to_persist[column_name] = str(val)
+            else:
+                timedelta_from_epoch = datetime.datetime.utcfromtimestamp(0) + val
+                row_to_persist[column_name] = timedelta_from_epoch.isoformat() + '+00:00'
```

Comment:
in what condition this could happen? What else the `property_format` theoretically could be if not `time`?

Reply:
Good question!
So when you run a select with `pymysql` to fetch a column of type `time`, it's mapped to python's `datetime.timedelta` (it could have been mapped to `datetime.time` imo, but they must have a reason :shrug:). I have this condition because I'm not sure if there is some other sql type that's also mapped to `datetime.timedelta`  thus the format could be null for example.